<a href="https://colab.research.google.com/github/Daniel-B-Oliveira/Estatistica-Computacional/blob/main/Lista01_p2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
!pip install scipy

In [ ]:
import pandas as pd
import scipy as scipy
from scipy import stats
import numpy as np

#**Questão 1**
Utilize o banco de dados Usinagem.eixos e teste a hipótese de que a média geral (independente do tipo de usinagem) da variável medida é diferente de 20 ($H_0 : µ = 20$ vs $H_a : µ \neq 20 $) a 5% de significância. Realize este teste utilizando o Python. Admita que o pressuposto de normalidade seja atendido. Apresente uma conclusão.


In [ ]:
usinagem = pd.read_csv("Usinagem.eixos.txt", sep="\t", decimal=",")
usinagem.head()

Sist-Usinagem   Medida
0              1  18.7997
1              1  20.5035
2              1  18.6214
3              1  19.9192
4              1  21.1170

In [ ]:
#Adimitindo normalidade
stats.ttest_1samp(usinagem['Medida'], popmean=20)

TtestResult(statistic=4.56966292248483, pvalue=2.879416168496031e-05, df=54)

Existem evidências suficientes para concluir que a média da variável "Medida" é significativamente diferente de 20, a 5% de significância.

#**Questão 2**
Utilize o banco de dados telecom users e teste a hipótese de que mulheres não diferem, em média, quanto à variável total gasto. Utilize 5% de significância. Realize este teste utilizando o Python. Admita que o pressuposto de normalidade seja atendido. Verifique a hipótese de igualdade das variâncias e utilize este resultado para a aplicação do teste t para comparação do dois grupos em média. Apresente uma conclusão.


In [ ]:
telecom = pd.read_csv("telecom_users.txt", sep=';', decimal=',')
telecom.head()

Codigo   IDCliente     Genero  Aposentado Casado Dependentes  \
0    1869  7010-BRBUU  Masculino           0    Sim         Sim   
1    4528  9688-YGXVR   Feminino           0    Nao         Nao   
2    6344  9286-DOJGF   Feminino           1    Sim         Nao   
3    6739  6994-KERXL  Masculino           0    Nao         Nao   
4     432  2181-UAESM  Masculino           0    Nao         Nao   

   MesesComoCliente ServicoTelefone MultiplasLinhas ServicoInternet  ...  \
0                72             Sim             Sim             Nao  ...   
1                44             Sim             Nao           Fibra  ...   
2                38             Sim             Sim           Fibra  ...   
3                 4             Sim             Nao             DSL  ...   
4                 2             Sim             Nao             DSL  ...   

  ProtecaoEquipamento ServicoSuporteTecnico ServicoStreamingTV ServicoFilmes  \
0         SemInternet           SemInternet        SemInternet   SemInternet   
1                 Sim                   Nao                Sim           Nao   
2                 Nao                   Nao                Nao           Nao   
3                 Nao                   Nao                Nao           Sim   
4                 Sim                   Nao                Nao           Nao   

  TipoContrato FaturaDigital    FormaPagamento ValorMensal TotalGasto Churn  
0       2 anos           Nao     CartaoCredito        24.1    1734.65   Nao  
1       Mensal           Sim     CartaoCredito       88.15     3973.2   Nao  
2       Mensal           Sim  DebitoAutomatico       74.95    2869.85   Sim  
3       Mensal           Sim  BoletoEletronico        55.9      238.5   Nao  
4       Mensal           Nao  BoletoEletronico       53.45      119.5   Nao  

[5 rows x 22 columns]

In [ ]:
telecom = telecom.dropna(subset=['TotalGasto'])
telecom['TotalGasto'] = pd.to_numeric(telecom['TotalGasto'], errors="coerce")

In [ ]:
feminino = telecom[telecom['Genero'] == 'Feminino']['TotalGasto']
masculino = telecom[telecom['Genero'] == 'Masculino']['TotalGasto']

In [ ]:
def f_test(group1, group2):

  #ddof = 1: variância amostral
  #ddof = 0: variância populacional

  #https://numpy.org/devdocs/reference/generated/numpy.var.html#numpy.var

  f = np.var(group1, ddof=1)/np.var(group2, ddof=1)
  nun = group1.size - 1
  dun = group2.size - 1
  p_value = 2*min([1-stats.f.cdf(f, nun, dun), stats.f.cdf(f, nun, dun)])
  return f, p_value

In [ ]:
f_test(feminino, masculino)

(1.0354031313628973, 0.3417153091169307)

Como o p-valor (0.3417) é maior que o nível de significância de 5%, não rejeitamos a hipótese nula. Isso significa que não há evidências suficientes para concluir que as variâncias dos grupos "feminino" e "masculino" são significativamente diferentes.

In [ ]:
stats.ttest_ind(feminino, masculino, equal_var=False)

TtestResult(statistic=0.2599236489868282, pvalue=0.7949316530053292, df=5956.060743603287)

Com base nos resultados dos testes realizados, o teste F apresentou um p-valor de 0.3417, indicando que não há evidências suficientes para concluir que as variâncias dos grupos "feminino" e "masculino" são significativamente diferentes. Além disso, o teste t apresentou um p-valor de 0.7949, sugerindo que não há diferença significativa entre as médias dos grupos. Em ambos os casos, como os p-valores são maiores que o nível de significância de 5% (𝛼=0.05), não rejeitamos as respectivas hipóteses nulas, ou seja, as variâncias e médias dos grupos são estatisticamente semelhantes.

#**Questão 3**

Um estudo com o objetivo de avaliar a efetividade de uma dieta combinada com um programa de exercícios físicos na redução do nível de colesterol foi realizado e os dados são como a seguir:

#### Tabela. Níveis de colesterolantes e após a dieta.

| **Antes** | **Depois** |
|-----------|------------|
| 201       | 200        |
| 231       | 236        |
| 221       | 216        |
| 260       | 233        |
| 228       | 224        |
| 237       | 216        |
| 326       | 296        |
| 319       | 305        |
| 245       | 232        |
| 240       | 207        |
| 267       | 247        |
| 284       | 210        |
| 201       | 209        |

Utilize 10% de significância. Realize este teste utilizando o Python. Admita que o pressuposto de normalidade seja atendido. A dieta foi significativamente eficiente para a redução dos níveis de colesterol? Justifique.


In [ ]:
dieta = pd.read_csv("dieta_q3_l1_p2.txt", sep=';')
dieta.head()

Antes  Depois
0    201     200
1    231     236
2    221     216
3    260     233
4    228     224

In [ ]:
d = dieta['Depois'] - dieta['Antes']

In [ ]:
stats.ttest_1samp(d, popmean=0, alternative='less')

TtestResult(statistic=-3.020113428672492, pvalue=0.005826676227150722, df=11)

O teste t para amostras pareadas foi realizado para avaliar a efetividade de uma dieta combinada com um programa de exercícios físicos na redução dos níveis de colesterol. Com um nível de significância de 10%, rejeitamos a hipótese nula. Portanto, há evidências estatísticas significativas para concluir que a dieta foi efetiva na redução dos níveis de colesterol, indicando que a média dos níveis de colesterol depois da dieta é significativamente diferente da média dos níveis antes da dieta.

#**Questão 4**
Utilize o banco de dados telecom users e teste a hipótese de que a probabilidade do cliente escolher qualquer uma das formas de pagamento é a mesma (distribuição igual nas diferentes formas de pagamento). Utilize 5%
de significância. Realize este teste utilizando o Python. Apresente uma conclusão.


In [ ]:
tabela4 = telecom['FormaPagamento'].value_counts()
esperado4 = tabela4.sum()*np.array(([0.25,0.25,0.25,0.25]))

In [ ]:
stats.chisquare(f_obs=tabela4, f_exp=esperado4)

Power_divergenceResult(statistic=233.0912128299365, pvalue=2.967840843314432e-50)

 O teste qui-quadrado de aderência foi realizado para avaliar se a probabilidade dos clientes escolherem qualquer uma das formas de pagamento é a mesma. Com um nível de significância de 5%, há evidências estatísticas significativas para concluir que a distribuição das formas de pagamento não é igual entre os clientes.

#**Questão 5**
Utilize o banco de dados telecom users e teste a hipótese de que forma de pagamento e gênero são variáveis dependentes. Utilize 10% de significância. Realize este teste utilizando o Python. Apresente uma conclusão.


In [ ]:
tabela5 = pd.crosstab(telecom['FormaPagamento'], telecom['Genero'])

In [ ]:
stats.chi2_contingency(tabela5)

Chi2ContingencyResult(statistic=1.8789066555627512, pvalue=0.5979162865536407, dof=3, expected_freq=array([[ 983.89842967, 1022.10157033],
       [ 671.46408286,  697.53591714],
       [ 639.09254928,  663.90745072],
       [ 641.54493819,  666.45506181]]))

O teste qui-quadrado de independência foi realizado para avaliar se a forma de pagamento e o gênero são variáveis dependentes. Com um nível de significância de 10%, não há evidências estatísticas suficientes para concluir que a forma de pagamento e o gênero são variáveis dependentes. Isso indica que as escolhas de forma de pagamento dos clientes não estão relacionadas ao gênero.

#**Questão 6**
Utilize o banco de dados telecom users e teste a hipótese de que forma de pagamento e cancelamento (churn) são variáveis dependentes. Utilize 5% de significância. Realize este teste utilizando o Python. Apresente uma conclusão

In [ ]:
tabela6 = pd.crosstab(telecom['FormaPagamento'], telecom['Churn'])

In [ ]:
stats.chi2_contingency(tabela6)

Chi2ContingencyResult(statistic=532.6218703571735, pvalue=4.0606265181460287e-115, dof=3, expected_freq=array([[1474.08320802,  531.91679198],
       [1005.25714286,  362.74285714],
       [ 957.49273183,  345.50726817],
       [ 961.16691729,  346.83308271]]))

O teste qui-quadrado de independência foi realizado para avaliar se a forma de pagamento e o cancelamento são variáveis dependentes. Com um nível de significância de 5%, há evidências estatísticas suficientes para concluir que a forma de pagamento e cancelamento são variáveis dependentes. Isso indica que as escolhas de forma de pagamento dos clientes estão significatimente relacionadas ao cancelamento.



#**Questão 7**
Teste as hipóteses da 1ª, 2ª e 3ª Questão desta lista utilizando o teste de Wilcoxon ou de Mann-Whitney (testes não paramétricos - não necessitam da suposição de normalidade). Realize estes testes no R. Apresente uma conclusão.
Verifique as hipóteses da 2ª e da 3ª Questão, por meio do teste de Wilcoxon ou Mann-Whtiney, também no Python.

In [ ]:
stats.mannwhitneyu(feminino, masculino)

MannwhitneyuResult(statistic=4468895.0, pvalue=0.9236419117127317)

In [ ]:
stats.wilcoxon(feminino, masculino)

KruskalResult(statistic=0.009188127702517866, pvalue=0.923635955645812)

Com um nível de significância de 5%, não há evidências estatísticas suficientes para concluir que mulheres e homens diferem em média quanto ao total gasto. Isso indica que as medianas de total gasto entre os gêneros são semelhantes na amostra analisada.

In [ ]:
stats.wilcoxon(dieta['Depois'], dieta['Antes'], alternative='less')

WilcoxonResult(statistic=8.5, pvalue=0.006103515625)

#**Questão 8**
A hipótese de que o total gasto é o mesmo nas diferentes formas de pagamento deve ser rejeitada? Realize o teste apropriado para responder a essa questão, supondo que os pressupostos necessários para o uso de testes paramétricos não sejam atendidos. Use 5% de significância. Realize este teste tanto utilizando o R quanto o Python. Apresente uma conclusão.

In [ ]:
dados8 = [telecom[telecom['FormaPagamento'] == forma]['TotalGasto']
          for forma in telecom['FormaPagamento'].unique()]

In [ ]:
stats.kruskal(*dados8)

KruskalResult(statistic=899.2210723877122, pvalue=1.3061125097862749e-194)

O teste de Kruskal-Wallis foi realizado para avaliar se há diferenças significativas no total gasto entre as diferentes formas de pagamento. Com um nível de significância de 5%, há evidências estatísticas suficientes para concluir que o total gasto varia significativamente de acordo com a forma de pagamento utilizada. Isso indica que as escolhas de forma de pagamento dos consumidores estão significativamente relacionadas ao valor gasto, sugerindo que diferentes métodos de pagamento influenciam o comportamento de gasto dos clientes.